torch/_decomp/decompositions.py

```diff
@@ -1840,6 +1840,111 @@ def _native_batch_norm_legit_functional(
     return output, save_mean, save_rstd, new_running_mean, new_running_var
 
 
+def _get_batch_norm_reserve_tensor(
```

Comment:
It might be worth a comment/note here describing what the result of tracing will give you when using compile/export here. Something about how we unconditionally return this extra reserve tensor (needed by cudnn, which export needs to be able to call for numeric accuracy), but only give it the correct shape in the traced graph if we detect that we would have called cudnn in eager mode (and export / compiler backends that do not wish to use cudnn are free to use the `new_batch_norm_with_stats_functional` decomp to decompose this op and rely on DCE to avoid materializing this tensor)